src/main/java/org/restcomm/sdk/RestcommClient.java

```diff
@@ -1,48 +1,99 @@
 package org.restcomm.sdk;
 
-import org.restcomm.sdk.domain.*;
-import org.restcomm.sdk.endpoints.RestEndpoints;
-
+import java.util.HashMap;
 import java.util.Map;
 
+import org.restcomm.sdk.domain.Account;
+import org.restcomm.sdk.domain.Application;
+import org.restcomm.sdk.domain.CallPage;
+import org.restcomm.sdk.domain.Client;
+import org.restcomm.sdk.domain.ExtensionData;
+import org.restcomm.sdk.domain.IncomingPhoneNumber;
+import org.restcomm.sdk.domain.RecordingPage;
+import org.restcomm.sdk.domain.ShortMessage;
+import org.restcomm.sdk.domain.XmppMapping;
+import org.restcomm.sdk.endpoints.RestEndpoints;
+
 /**
  * @author oleg.agafonov@telestax.com (Oleg Agafonov)
+ * @author sergey vetyutnev
  */
 public class RestcommClient {
 
     private final HttpClient httpClient;
 
-    private final String baseRestcommUrl;
-    private final String baseRestcommUrlWithoutAccount;
+    protected final String baseRestcommUrl;
+    protected final String baseRestcommUrlWithoutAccount;
 
-    private final String baseUrl;
+    protected final String baseUrl;
+    private final String accountSid;
 
-    private final Map<String, String> endpoints;
+    protected final Map<String, String> endpoints;
 
     public RestcommClient(RestcommClientConfiguration config) {
         this.httpClient = new HttpClient(config.getAccountSid(), config.getAccountToken());
-        this.baseRestcommUrlWithoutAccount = config.getBaseUrl() + "/restcomm/2012-04-24/Accounts";
-        this.baseRestcommUrl = config.getBaseUrl() + "/restcomm/2012-04-24/Accounts/" + config.getAccountSid();
+
         this.baseUrl = config.getBaseUrl();
-        this.endpoints = config.getEndpoints();
+        this.baseRestcommUrlWithoutAccount = baseUrl + "/restcomm/2012-04-24/Accounts";
+        this.baseRestcommUrl = baseUrl + "/restcomm/2012-04-24/Accounts/" + config.getAccountSid();
+        if (config.getEndpoints() != null)
+            this.endpoints = config.getEndpoints();
+        else
+            this.endpoints = new HashMap<String, String>();
```

Comment:
Consider `Collections.emptyMap()`

Reply:
Collections.emptyMap() gives us an immutable collection but we will add elemets to collection by a code logic. IMO Collections.emptyMap() is not an option. CAn you comment ?

Reply:
Ahh, not, its inoy changed, you are right